src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -39,129 +40,112 @@
    */
   public void distributeSampleUnits(CollectionExercise exercise) {
 
+    List<ExerciseSampleUnitGroup> sampleUnitGroups = new ArrayList<>();
     try {
-      List<ExerciseSampleUnitGroup> sampleUnitGroups = retrieveSampleUnitGroups(exercise);
-
-      for (ExerciseSampleUnitGroup anExerciseSampleUnitGroup : sampleUnitGroups) {
-        distributeSampleUnits(exercise, anExerciseSampleUnitGroup);
-      }
-
-      if (!sampleUnitGroups.isEmpty()) {
-        collectionExerciseTransitionState(exercise);
-      }
-
-    } catch (LockingException | CTPException ex) {
-      log.error("Distribution failed", ex);
-    } finally {
-      try {
-        sampleDistributionListManager.deleteList(DISTRIBUTION_LIST_ID, true);
-      } catch (LockingException ex) {
-        log.error("Failed to release sampleDistributionListManager data", ex);
-      }
+      sampleUnitGroups = retrieveSampleUnitGroups(exercise);
+    } catch (LockingException ex) {
+      log.error("Sample Unit Distribution failed", ex);
     }
-  }
-
-  /**
-   * Distribute SampleUnits for a SampleUnitGroup.
-   *
-   * @param exercise CollectionExercise of which sampleUnitGroup is a member.
-   * @param sampleUnitGroup for which to distribute sample units.
-   */
-  private void distributeSampleUnits(
-      CollectionExercise exercise, ExerciseSampleUnitGroup sampleUnitGroup) throws CTPException {
-    List<ExerciseSampleUnit> sampleUnits = sampleUnitRepo.findBySampleUnitGroup(sampleUnitGroup);
-    List<SampleUnit> children = new ArrayList<>();
-    SampleUnitParent parent = null;
-    for (ExerciseSampleUnit sampleUnit : sampleUnits) {
-      String activeActionPlanId =
-          sampleUnitDistributorHelper.getActiveActionPlanId(
-              exercise.getExercisePK(),
-              sampleUnit.getSampleUnitType().name(),
-              exercise.getSurveyId());
-      if (sampleUnit.getSampleUnitType().isParent()) {
-        parent = sampleUnit.toSampleUnitParent(activeActionPlanId, exercise.getId());
-      } else {
-        SampleUnit child = sampleUnit.toSampleUnitChild(activeActionPlanId);
-        children.add(child);
-      }
-    }
-
-    if (parent == null) {
-      log.with("sample_unit_group_pk", sampleUnitGroup.getSampleUnitGroupPK())
-          .error("No Parent for SampleUnit in SampleUnitGroup");
+    if (sampleUnitGroups.isEmpty()) {
+      log.with("collection_exercise_id", exercise.getId())
+          .debug("No sample unit groups to distribute for exercise");
       return;
     }
 
-    if (!children.isEmpty()) {
-      parent.setSampleUnitChildren(new SampleUnitChildren(children));
-    }
+    // Catch errors distributing sample units so that only failing units are stopped
+    sampleUnitGroups.forEach(
+        sampleUnitGroup -> {
+          try {
+            distributeSampleUnitGroup(exercise, sampleUnitGroup);
+          } catch (CTPException ex) {
+            log.with("sampleUnitGroupPK", sampleUnitGroup.getSampleUnitGroupPK())
```

Comment:
I am deeply dubious about catches which do not rethrow. How will anything be retried?

Reply:
I think failing sample units would be retried on the next schedule, I'm having a look at sampleUnitDistributor more generally now so will see if I can find any improvements